h2o-core/src/main/java/hex/grid/Grid.java

```diff
@@ -287,10 +287,10 @@ private void appendFailedModelParameters(final Key<Model> modelKey, final MP par
    * @param params model parameters which caused model builder failure
    * @params e  exception causing a failure
    */
-  void appendFailedModelParameters(final Key<Model> modelKey, final MP params, final Exception e) {
+  void appendFailedModelParameters(final Key<Model> modelKey, final MP params, final Throwable t) {
```

Comment:
is it necessary? are you catching Java Errors somewhere? It's not recommended in most cases...

Reply:
A little bit short-sighted of me. I used `Throwable`, because ModelBuilder's `onExceptionalCompletion` method uses `Throwable` as well - even though it might be a little bit more justified there. 

Yes, errors are nowhere caught as far as I can see in the model builders. But what if ? What if the model builder fails and calls `onExceptionalCompletion` with an `Error` (not an `Exception`) as an argument ? Shall we discard the error ? Basically, the job may fail due to the error, or it might be caught somewhere..but this method is about telling the grid which parameters failed and why. And the `why` is typed as `Throwable` in ModelBuilder.

1. Filter out errors and only pass exceptions ?
2. Modify ModelBuiler ?
3. Something else ?

Reply:
Error should never be caught, we should use Exception here

Reply:
We are not actually catching Throwable, this comes from CountedCompleter - https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CountedCompleter.html

I think for now keeping the Throwable is fine - @Pscheidl is this a user-facing or just an internal API?

Reply:
Internal. Appending the failed results happens inside GridSearch.

Reply:
I am unable to explain this originates from core and in order to change it, we must take one of the actions mentioned above.

Reply:
As long as this is something we can change in the future without affecting users - I think we do not have to address it in this PR. Existing GridSearch catches Throwable - this should be revisited in a separate activity, not in the scope of this PR.

Reply:
Great, thank you for understanding. Let's no address refactorings in this PR whatsoever - these are hot topics. @seb-h2o please create a JIRA to address it if you want `Throwable` to be gone, or I can create one, no problemo.

I only agree the Throwable is weird. But it's honoring existing contract.